packages/interactivity/docs/2-api-reference.md

```diff
@@ -0,0 +1,546 @@
+# API Reference
+
+To add interactivity to blocks using the Interactivity API, developers can use:
+
+- **Directives** - added to the markup to add specific behavior to the DOM elements of block.
+- **Store** - that contains the logic and data (state, actions, or effects among others) needed for the behaviour.
+
+DOM elements are connected to data stored in the state & context through directives. If data in the state or context change, directives will react to those changes updating the DOM accordingly (see [diagram](https://excalidraw.com/#room=11c461f5e18480cd8631,mNsrOHbcUKgVdITiRl6H5w)).
+
+![State & Directives](assets/state-directives.png)
+
+## The directives
+
+Directives are custom attributes that are added to the markup of your block to add behaviour to its DOM elements. This can be done in the `render.php` file (for dynamic blocks) or the `save.js` file (for static blocks).
+
+Interactivity API directives use the `data-` prefix.
+
+_Example of directives used in the HTML markup_
+
+```html
+<div
+  data-wp-context='{ "isOpen": false }'
+  data-wp-effect="effects.logIsOpen"
+>
+  <button
+    data-wp-on--click="actions.toggle"
+    data-wp-bind--aria-expanded="context.isOpen"
+    aria-controls="p-1"
+  >
+    Toggle
+  </button>
+ 
+  <p id="p-1" data-wp-show="context.isOpen">
+    This element is now visible!
+  </p>
+</div>
+```
+
+### List of Directives
+
+With directives we can manage directly in the DOM behavior related to things such as: Side Effects, State, Event Handlers, Attributes, Display, Template Logic, Content or Errors
+
+#### `wp-effect` ![](https://img.shields.io/badge/SIDE_EFFECTS-207399.svg)
+
+It runs an expression **when the node is created and runs it again when the state or context changes**. You can call several effects (or inits) from the same DOM by using the syntax`data-effect--[unique-id]`
+
+_Example of `wp-effect` directive_
+```html
+<form 
+  data-wp-effect="effect.myNamespace.logTime" 
+>
+  <input type="text" id="password" name="access-password">
+</form>
+```
+
+_Example of several `wp-effect` directives on the same DOM element_
+```html
+<form 
+  data-wp-effect–-1="effect.myNamespace.logTime" 
+  data-wp-effect–-2="effect.myNamespace.focusFirstElement"
+>
+  <input type="text" id="password" name="access-password">
+</form>
+```
+
+<details>
+  <summary><em>See store used with the directive above</em></summary>
+<pre><code>
```

Comment:
It's the only way I found to add a `details` behavior in GitHub's markdown (I want to "hide" the store examples but let them available as they're useful to understand the directives behaviour)

Reply:
I think the details are fine. I meant using markdown's ```js instead of `<pre><code>` so GitHub's code highlighter works. 

Like this (but without the initial `\` in the ```js):

```md
<details>
  <summary><em>See store used with the directive above</em></summary>

\```js
// store
store({
  effects: {
    myNamespace: {
      logTime: () => console.log(new Date()),
      focusFirstElement: ({ref}) => ref.querySelector('input:first-child').focus();
    },
  }
});
\```
</details>
```

Full example here: https://gist.github.com/luisherranz/0a065f2862596903bbef0c7d56428662

Reply:
Wow! I was looking for a way to do this. Thanks! 
Yes, with this reference I will add these code blocks in this way